posts/avoid-timeouts-azure-data-factory.md

```diff
@@ -0,0 +1,201 @@
+---
+authors:
+- Sam Portillo
+tags:
+- azure
+- big-data
+- data-engineering
+date: 2022-01-07T12:21:50.000Z
+title: "Avoid server-side timeouts when copying data from Azure Data Factory"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2022/01/adf-timeouts-cover.png
+---
+
+## Introduction
+This article walks through an ingestion strategy called iterative range copy to bulk load SQL tables using Azure Data Factory (ADF). The major benefits of this approach are avoiding server-side timeouts and controlling the size of batch copies.  
+
+This strategy only uses ADF native functions, no external services, tools, or code is required.  
+<br/>
+
+## High-level overview
+This strategy uses an ADF “**Until**” function that loops through a table at a specific row range at a time. The “Until” function is a special type of loop that continues to run until a condition is met. 
+
+Each iteration copies the rows from the given row range and writes them to a sink.For example, if you are copying 1,000 rows at a time from a table, the first iteration would copy rows from indexes 1 to 1,000, then the second iteration would copy rows 1,001 to 2,000 and so on.  
+Since the row range is set to a size of 1,000 the “Until” function will stop after a set of results returning less than 1,000 rows has been processed. This means there are no more rows in the source table to copy.  
```

Comment:
```suggestion
Since the row range is set to a size of 1,000, the “Until” function will stop after a set of results returning _less_ than 1,000 rows has been processed. This means there are no more rows in the source table to copy.  
```